agentMET4FOF/agents.py

```diff
@@ -456,18 +456,30 @@ def send_plot(self, fig=plt.Figure()):
         fig : Figure
             Can be either matplotlib figure or plotly figure
 
+        mode : str
+            "image" - converts into image via encoding at base64 string.
+            "plotly" - converts into plotly figure using `mpl_to_plotly`
+            Default: "image"
+
         Returns
         -------
         The message format is {'from':agent_name, 'plot': data, 'senderType': agent_class}.
         """
+
         if isinstance(fig, matplotlib.figure.Figure):
-            #graph = self._convert_to_plotly(fig) #unreliable
-            graph = self._fig_to_uri(fig)
+            if mode == "plotly":
+                graph = self._convert_to_plotly(fig)
+            elif mode == "image":
+                graph = self._fig_to_uri(fig)
         elif isinstance(fig, dict): #nested
-            for key in fig.keys():
-                fig[key] = self._fig_to_uri(fig[key])
+            if mode == "plotly":
+                for key in fig.keys():
+                    fig[key] = self._convert_to_plotly(fig[key])
+            if mode == "image":
```

Comment:
We should use `elif` here as well to be consistent and clear or follow the decision we made about the input checks.